### PR TITLE
Excel: ensure that set column and row headers are remembered on copied sheets.

### LINKED ISSUE #6628: Excel: row/column headers should be remembered when copying sheets
Currently if row/column headers are set by NVDA on an Excel sheet, and that sheet is copied within the same workbook, NVDA does not detect the set row and column headers on the new copy. This is because NVDA stores defined names globally on the workbook, rather than per sheet.

### PR DESCRIPTION
**Pull request by michaelDCurran**
Fixes #6628

### CODE REVIEW

**Comment by michaelDCurran** [comment]
@qchristensen: when the appveyor build is complete for this branch, could you please give it a test, especially on Excel 2007 if you have it. Now column and row headers set by this build of NVDA or newer will be remembered when a sheet is copied.

File: source/NVDAObjects/window/excel.py [hunk 1]
```diff
@@ -670,7 +670,7 @@ def _get_excelApplicationObject(self):
 		self.excelApplicationObject=self.excelWorksheetObject.application
 		return self.excelApplicationObject
 
-	re_definedName=re.compile(ur'^((?P<sheet>\w+)!)?(?P<name>\w+)(\.(?P<minAddress>[a-zA-Z]+[0-9]+)?(\.(?P<maxAddress>[a-zA-Z]+[0-9]+)?(\..*)*)?)?$')
+	re_definedName=re.compile(u'^((?P<sheet>(\'[^\']+\'|[^!]+))!)?(?P<name>\w+)(\.(?P<minAddress>[a-zA-Z]+[0-9]+)?(\.(?P<maxAddress>[a-zA-Z]+[0-9]+)?(\..*)*)?)?$')
```
Comment (feerrenrut) [inline source/NVDAObjects/window/excel.py]: As we discussed could we try and split this up to comment each part of this (or at least comment what we can). I think examples of the edge cases that we are trying to match are quite helpful too. In the absence of automated tests, this at least gives us a suite of manual test cases to use the next time that we find we have to modify this.

**Review by feerrenrut — CHANGES_REQUESTED** [review]

**Comment by michaelDCurran** [comment]
@feerrenrut: I have made the requested change.

**Review by feerrenrut — APPROVED** [review]
